src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/InternalForeign.java

```diff
@@ -49,18 +49,18 @@ public static InternalForeign getInstancePrivileged() {
 
     @Override
     public MemoryAddress withSize(MemoryAddress base, long byteSize) throws IllegalAccessError {
-        return Utils.makeNativeSegmentUnchecked(base.toRawLongValue(), byteSize, null, false)
+        return NativeMemorySegmentImpl.makeNativeSegmentUnchecked(base.toRawLongValue(), byteSize, null, false)
```

Comment:
Pre-existing; should this check that `base` has the Nothing segment as segment? (otherwise access modes are dropped for instance).